tests/python/pants_test/engine/legacy/test_graph.py

```diff
@@ -78,7 +78,7 @@ def test_invalidate_fsnode_incremental(self):
 
       # Invalidate the '3rdparty/python' DirectoryListing, the `3rdparty` DirectoryListing,
       # and then the root DirectoryListing by "touching" files/dirs.
-      for filename in ('3rdparty/python/BUILD', '3rdparty/python', 'non_existing_file'):
+      for filename in ('3rdparty/python/BUILD', '3rdparty/python', ''):
```

Comment:
Oohoo!

So, I removed this case because generally a file changing should not invalidate its parent directory. But I forgot that this is here explicitly because the root is a special case: we will never receive a change notification for the root, so all changes to files under the root need to be treated as a change to the root.

Will fix and make that clearer in the test cases. Thanks!